.circleci/config.yml

```diff
@@ -0,0 +1,75 @@
+version: 2
+jobs:
+  test:
+    working_directory: /go/src/github.com/Azure/acs-engine
+    docker:
+      - image: microsoft/acs-engine:latest
+    steps:
+      - setup_remote_docker:
+          reusable: true
+      - checkout
+      - run:
+          name: Build current image
+          command: |
+            TAG="0.1.${CIRCLE_BUILD_NUM}"
+            docker build -t microsoft/acs-engine:$TAG .
```

Comment:
The only thing which this won't catch is upgrades to `GO_VERSION` or `AZURE_CLI_BUNDLE_VERSION` but IMO the test phase should be as quick as possible.  Currently, it is ~11 minutes. We could keep the build image and use it for e2e test which would catch this.